### ISSUE TITLE
Percentages per file should add up to 100%

### ISSUE THREAD

**Issue by davbeek** [davbeek/gitinspectorgui#56]
While grading, I noticed that the current way of calculating percentages for Lines and Insertions to sum all files to 100% is confusing when certain files (e.g. plant files) are much larger than other important files (controller).

This means that the output for Authors-Files and Files-Authors should be changed so that the total of percentages per file should be 100%. The output in Authors and Files can remain the same.

**Comment by Alberth289346** [comment]
> The problem however is that when there are no particular very large files, the current way of calculating the percentages works well. So perhaps we need an option for this?

If possible, avoiding having options is generally preferred.

I don't quite understand how it gets broken for "all files mostly the same size". If you have 2 files of equal line count, then basically editing a line in either file has the same weight, no matter how you count, as far as I can see. Why is that wrong?

**Comment by davbeek** [comment]
The output is wrong. The question is what is the most user-friendly, useful output. It will become clearer when looking at a concrete output example. The three colors are for three authors. I have removed the names because of privacy issues. The authors have not used any definitions/templates and instantiations, but just copied code for 3 pushers and 3 tubes and various products in file `plant.cif`, leading to huge files:
<img width="599" alt="image" src="https://github.com/user-attachments/assets/50c8b44e-7bb9-4796-9cb3-f3bcaa94aca7">

**Comment by davbeek** [comment]
I have made several edits in the previous item, you may need to refresh.

**Comment by Alberth289346** [comment]
Lots of numbers, I am not sure I understand them. I'd have to study them more closely, but not now.

I wonder if you wouldn't want a graphical layout instead. I am assuming the actual numbers are not that relevant, it's more about relative magnitude.

For a single file, a piechart would work (breakdown of each person that worked on the file). You can also make a piechart for a person (breakdown of how much they work on each file).

Maybe other graphical forms could work too.


[ probably in a next release :) ]

**Comment by davbeek** [comment]
Good point, you suggestion to add some graphical output. Indeed piecharts would work very well. For Excel it is trivial adding piecharts and for html, I expect piecharts should also be possible.

**Comment by davbeek** [comment]
Blames in the final result. E.g. for the file `plant.cif`, the top entry is 1088 lines, which equals the total number of lines in the file. This is not even the most extreme version. There are groups with a `plant.cif` file of 2000 or more lines. For other workstations, plant files are much smaller.

**Comment by Alberth289346** [comment]
Yeah, copy/paste is cheap if you don't need long-term maintenance or extensive changes.

I am not sure what to suggest otherwise here, maybe first wait for the results of piecharting it?  
Pie-charts move away from the precise numbers, and automatically scale to 100% (or rather, 360 degrees) :)
